js/integration-tests/solidity-tests/test/fuzz.ts

```diff
@@ -0,0 +1,105 @@
+import chai, { assert, expect } from "chai";
+import { TestContext } from "./testContext";
+import fs from "node:fs/promises";
+import { existsSync } from "node:fs";
+import { FuzzConfigArgs } from "@nomicfoundation/edr";
+
+describe("Fuzz and invariant testing", function () {
+  let testContext: TestContext;
+
+  before(async function () {
+    testContext = await TestContext.setup();
+  });
+
+  it("FailingFuzz", async function () {
+    const failureDir = testContext.fuzzFailuresPersistDir;
+
+    // Remove invariant config failure directory to make sure it's created fresh.
+    await fs.rm(failureDir, {
+      recursive: true,
+      force: true,
+    });
+
+    const result1 = await testContext.runTestsWithStats("OverflowTest");
+    assert.equal(result1.failedTests, 1);
+    assert.equal(result1.totalTests, 1);
+
+    // The fuzz failure directory should not be created if we don't set the directory
+    assert.isFalse(existsSync(failureDir));
+
+    const result2 = await testContext.runTestsWithStats("OverflowTest", {
+      fuzz: {
+        failurePersistDir: failureDir,
+      },
+    });
+    assert.equal(result2.failedTests, 1);
+    assert.equal(result2.totalTests, 1);
+
+    // The fuzz failure directory should now be created
+    assert.isTrue(existsSync(failureDir));
+  });
+
+  // One test as steps should be sequential
+  it("FailingInvariant", async function () {
+    const failureDir = testContext.invariantFailuresPersistDir;
+    const defaultConfig = {
```

Comment:
I would rename this to `invariantConfig`

Reply:
Thanks, fixed in [ab94d0b](https://github.com/NomicFoundation/edr/pull/658/commits/ab94d0b77756dbc8b1caa6fdadb7db918b2fe2f6)